main/src/library/MutMap.flix

```diff
@@ -433,7 +433,7 @@ namespace MutMap {
     ///
     /// The function `f` must be impure.
     ///
-    pub def foreach(f: (k, v) ~> Unit, m: MutMap[k, v]): Unit & Impure =
+    pub def foreach(f: (k, v) -> Unit & ef, m: MutMap[k, v]): Unit & ef =
```

Comment:
f should have effect `ef` but the whole function should be impure (because it touches mutable memory.)